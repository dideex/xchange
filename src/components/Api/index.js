import {noty} from '../common'

export class Api {
  constructor() {
    this.backend = 'http://localhost:3030'
    this.isNetworkError = false
  }

  post(endpoint, data = {}, token = null) {
    return fetch(`${this.backend}/api/${endpoint}`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
      body: JSON.stringify(data),
    }).then(response => response.json())
  }

  get(endpoint, params = '', token = null) {
    return fetch(`${this.backend}/api/${endpoint}${params}`, {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
    }).then(response => response.json())
  }
  
  errorEmitter = fn => ({err, errCode, ...data}) => {
    if (err) {
      this.isNetworkError = true
      console.error('errorEmitter ___ ', errCode)
      noty(err, 'error')
    } else fn(data)
  }
}

export default new Api()
