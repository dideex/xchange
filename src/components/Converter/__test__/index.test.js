import React from 'react'
import Component from '../index'
import {shallow} from 'enzyme'
import {Provider as MobxProvider} from 'mobx-react'

import {fakeCurrnecy} from '../../../helpers'
import {mountWrap} from '../../../helpers/router-intl-context'
import {wrapInTestContext} from '../../../helpers/dnd'
import CashStore from '../../../store/Cash'
import UserStore from '../../../store/User'

import Api from '../../Api'

jest.mock('../../../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

jest.mock('../../../components/common/Noty.js', () => ({
  noty: () => {},
}))

describe('Converter index', () => {
  let cashStore = new CashStore()
  let userStore = new UserStore()
  Api.get = () => Promise.resolve({data: fakeCurrnecy, userRate: 1})
  beforeAll(() => {
    cashStore = new CashStore()
    cashStore.userRate = 1
    cashStore.fetchCurrency = () => Promise.Resolve
    userStore = new UserStore()
  })
  describe('Markup', () => {
    it('Basic markup', () => {
      const wrapper = mountWrap(
        <MobxProvider cashStore={cashStore} userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('Loading component', () => {
      cashStore.loading = true
      const wrapper = mountWrap(
        <MobxProvider cashStore={cashStore} userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('Component behaviour',() => {
    it('Should fetch currnecy after mounting', () => {
      const fetchCurrency = jest.fn(() => Promise.resolve())
      cashStore.fetchCurrency = fetchCurrency
      const wrapper = mountWrap(
        <MobxProvider cashStore={cashStore} userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      expect(fetchCurrency).toHaveBeenCalledTimes(1)
    })
  })
})
