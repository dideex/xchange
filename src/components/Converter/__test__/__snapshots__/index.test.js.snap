// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter index Markup Basic markup 1`] = `
"<div class=\\"css-1xiihgh\\"><div class=\\"css-2103di\\"><div class=\\"left__input css-prvt8v\\"><h2 class=\\"css-1swl25x\\"><span>From</span></h2><div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>eth</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>eth</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"0\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></div><span class=\\"css-1sr8xmg\\">tether label</span></div></div><div class=\\"tip__block css-hpqevb\\"><span><span>Min:</span>NaN tether label</span></div></div><div class=\\"css-164cafj\\"><div class=\\"css-tfqvvf\\"><h2 class=\\"css-1swl25x\\"><span>Rate exchange</span></h2><div class=\\"css-1nd2432\\"><span>-NaN</span><span>-NaN</span></div></div></div><div class=\\"right__input css-prvt8v\\"><h2 class=\\"css-1swl25x\\"><span>To</span></h2><div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>rur</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>rur</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"0\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></div><span class=\\"css-1sr8xmg\\">litecoin label</span></div></div><div class=\\"tip__block css-hpqevb\\"><span><span>Reserve:</span>NaN litecoin label</span></div></div></div><div class=\\"css-1lppr29\\"><div class=\\"css-adjgff\\"><svg viewBox=\\"0 0 38 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #aee239;\\" d=\\"M37.4832 32.8537L30.0829 25.4548C29.749 25.1208 29.2962 24.9353 28.8212 24.9353H27.6115C29.6599 22.3154 30.8771 19.0208 30.8771 15.436C30.8771 6.90902 23.9666 0 15.4389 0C6.91122 0 0 6.90902 0 15.436C0 23.963 6.91047 30.872 15.4389 30.872C19.0237 30.872 22.3197 29.6548 24.9396 27.6064V28.8162C24.9396 29.2912 25.1251 29.7439 25.4591 30.0779L32.8594 37.4768C33.5571 38.1744 34.6852 38.1744 35.3755 37.4768L37.4758 35.3764C38.1735 34.6787 38.1735 33.5513 37.4832 32.8537ZM15.4389 26.122C9.53558 26.122 4.75146 21.3468 4.75146 15.436C4.75146 9.5334 9.52741 4.75 15.4389 4.75C21.3422 4.75 26.1264 9.52523 26.1264 15.436C26.1264 21.3386 21.3504 26.122 15.4389 26.122ZM17.451 14.8007L14.1104 13.7987C13.7274 13.6837 13.4595 13.2955 13.4595 12.8539C13.4595 12.3144 13.8529 11.875 14.3353 11.875H16.4216C16.76 11.875 17.0866 11.9707 17.373 12.1511C17.6135 12.3018 17.9193 12.2929 18.1249 12.0969L18.9969 11.2649C19.2589 11.0148 19.2441 10.5814 18.9546 10.3639C18.2792 9.85699 17.4643 9.56457 16.6264 9.52152V8.3125C16.6264 7.98445 16.3607 7.71875 16.0327 7.71875H14.8452C14.5171 7.71875 14.2514 7.98445 14.2514 8.3125V9.50891C12.4976 9.55566 11.0838 11.0341 11.0838 12.8539C11.0838 14.3361 12.0479 15.6602 13.4276 16.0743L16.7682 17.0763C17.1511 17.1913 17.4191 17.5795 17.4191 18.0211C17.4191 18.5606 17.0257 19 16.5433 19H14.4577C14.1193 19 13.7927 18.9043 13.5063 18.7239C13.2658 18.5732 12.96 18.5821 12.7544 18.7781L11.8824 19.6101C11.6204 19.8602 11.6352 20.2936 11.9247 20.5111C12.6001 21.018 13.415 21.3104 14.2529 21.3535V22.5625C14.2529 22.8905 14.5186 23.1562 14.8466 23.1562H16.0341C16.3622 23.1562 16.6279 22.8905 16.6279 22.5625V21.3661C18.3817 21.3193 19.7955 19.8416 19.7955 18.0211C19.7955 16.5389 18.8314 15.2148 17.451 14.8007Z\\"></path></svg><input type=\\"text\\" placeholder=\\"Find\\" class=\\"css-487mdy\\" value=\\"\\"></div><div class=\\"css-eex5l9\\"><div class=\\"css-1i7yuyg\\"><div data-testid=\\"currency-badge-item\\" class=\\"css-17ixpwd\\"><div data-testid=\\"dragged-anchor\\" style=\\"display: inline-block;\\" draggable=\\"true\\"><span style=\\"cursor: move; z-index: 1; border-radius: 10px; color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity: 1;\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill: #aee239;\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill: #8fbe00;\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>eth</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>eth</text></svg><span style=\\"padding-left: 45px;\\">tether</span></span></div></div><div data-testid=\\"currency-badge-item\\" class=\\"css-17ixpwd\\"><div data-testid=\\"dragged-anchor\\" style=\\"display: inline-block;\\" draggable=\\"true\\"><span style=\\"cursor: move; z-index: 1; border-radius: 10px; color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity: 1;\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill: #aee239;\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill: #8fbe00;\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg><span style=\\"padding-left: 45px;\\">lisk</span></span></div></div><div data-testid=\\"currency-badge-item\\" class=\\"css-17ixpwd\\"><div data-testid=\\"dragged-anchor\\" style=\\"display: inline-block;\\" draggable=\\"true\\"><span style=\\"cursor: move; z-index: 1; border-radius: 10px; color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity: 1;\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill: #aee239;\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill: #8fbe00;\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>rur</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>rur</text></svg><span style=\\"padding-left: 45px;\\">litecoin</span></span></div></div></div></div></div><div class=\\"css-kszx3i\\"><label class=\\"css-gudk1o\\"><input placeholder=\\"Full name\\" pattern=\\"[a-zа-яё]{2,}\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input mask=\\"tether label\\" placeholder=\\"Your tether wallet\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input mask=\\"litecoin label\\" placeholder=\\"Your wallet for litecoin\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><div class=\\"css-75t9v1\\"><label class=\\"\\" for=\\"agree\\"><input id=\\"agree\\" type=\\"checkbox\\"><span>I agree to the processing of personal data</span></label></div><div class=\\"css-1xzpagd\\"><button cursor=\\"not-allowed\\" class=\\"css-zfl023\\"><div class=\\"css-1v1d0dg\\">Create</div></button></div></div></div>"
`;

exports[`Converter index Markup Loading component 1`] = `
"<div class=\\"css-1xiihgh\\"><div class=\\"css-2103di\\"><div class=\\"left__input css-prvt8v\\"><h2 class=\\"css-1swl25x\\"><span>From</span></h2><div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>eth</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>eth</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"0\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></div><span class=\\"css-1sr8xmg\\">tether label</span></div></div><div class=\\"tip__block css-hpqevb\\"><span><span>Min:</span>NaN tether label</span></div></div><div class=\\"css-164cafj\\"><div class=\\"css-tfqvvf\\"><h2 class=\\"css-1swl25x\\"><span>Rate exchange</span></h2><div class=\\"css-1nd2432\\"><span>-NaN</span><span>-NaN</span></div></div></div><div class=\\"right__input css-prvt8v\\"><h2 class=\\"css-1swl25x\\"><span>To</span></h2><div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>rur</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>rur</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"0\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></div><span class=\\"css-1sr8xmg\\">litecoin label</span></div></div><div class=\\"tip__block css-hpqevb\\"><span><span>Reserve:</span>NaN litecoin label</span></div></div></div><div class=\\"css-1lppr29\\"><div class=\\"css-adjgff\\"><svg viewBox=\\"0 0 38 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #aee239;\\" d=\\"M37.4832 32.8537L30.0829 25.4548C29.749 25.1208 29.2962 24.9353 28.8212 24.9353H27.6115C29.6599 22.3154 30.8771 19.0208 30.8771 15.436C30.8771 6.90902 23.9666 0 15.4389 0C6.91122 0 0 6.90902 0 15.436C0 23.963 6.91047 30.872 15.4389 30.872C19.0237 30.872 22.3197 29.6548 24.9396 27.6064V28.8162C24.9396 29.2912 25.1251 29.7439 25.4591 30.0779L32.8594 37.4768C33.5571 38.1744 34.6852 38.1744 35.3755 37.4768L37.4758 35.3764C38.1735 34.6787 38.1735 33.5513 37.4832 32.8537ZM15.4389 26.122C9.53558 26.122 4.75146 21.3468 4.75146 15.436C4.75146 9.5334 9.52741 4.75 15.4389 4.75C21.3422 4.75 26.1264 9.52523 26.1264 15.436C26.1264 21.3386 21.3504 26.122 15.4389 26.122ZM17.451 14.8007L14.1104 13.7987C13.7274 13.6837 13.4595 13.2955 13.4595 12.8539C13.4595 12.3144 13.8529 11.875 14.3353 11.875H16.4216C16.76 11.875 17.0866 11.9707 17.373 12.1511C17.6135 12.3018 17.9193 12.2929 18.1249 12.0969L18.9969 11.2649C19.2589 11.0148 19.2441 10.5814 18.9546 10.3639C18.2792 9.85699 17.4643 9.56457 16.6264 9.52152V8.3125C16.6264 7.98445 16.3607 7.71875 16.0327 7.71875H14.8452C14.5171 7.71875 14.2514 7.98445 14.2514 8.3125V9.50891C12.4976 9.55566 11.0838 11.0341 11.0838 12.8539C11.0838 14.3361 12.0479 15.6602 13.4276 16.0743L16.7682 17.0763C17.1511 17.1913 17.4191 17.5795 17.4191 18.0211C17.4191 18.5606 17.0257 19 16.5433 19H14.4577C14.1193 19 13.7927 18.9043 13.5063 18.7239C13.2658 18.5732 12.96 18.5821 12.7544 18.7781L11.8824 19.6101C11.6204 19.8602 11.6352 20.2936 11.9247 20.5111C12.6001 21.018 13.415 21.3104 14.2529 21.3535V22.5625C14.2529 22.8905 14.5186 23.1562 14.8466 23.1562H16.0341C16.3622 23.1562 16.6279 22.8905 16.6279 22.5625V21.3661C18.3817 21.3193 19.7955 19.8416 19.7955 18.0211C19.7955 16.5389 18.8314 15.2148 17.451 14.8007Z\\"></path></svg><input type=\\"text\\" placeholder=\\"Find\\" class=\\"css-487mdy\\" value=\\"\\"></div><div class=\\"css-eex5l9\\"><div class=\\"css-1i7yuyg\\"><div data-testid=\\"currency-badge-item\\" class=\\"css-17ixpwd\\"><div data-testid=\\"dragged-anchor\\" style=\\"display: inline-block;\\" draggable=\\"true\\"><span style=\\"cursor: move; z-index: 1; border-radius: 10px; color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity: 1;\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill: #aee239;\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill: #8fbe00;\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>eth</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>eth</text></svg><span style=\\"padding-left: 45px;\\">tether</span></span></div></div><div data-testid=\\"currency-badge-item\\" class=\\"css-17ixpwd\\"><div data-testid=\\"dragged-anchor\\" style=\\"display: inline-block;\\" draggable=\\"true\\"><span style=\\"cursor: move; z-index: 1; border-radius: 10px; color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity: 1;\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill: #aee239;\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill: #8fbe00;\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg><span style=\\"padding-left: 45px;\\">lisk</span></span></div></div><div data-testid=\\"currency-badge-item\\" class=\\"css-17ixpwd\\"><div data-testid=\\"dragged-anchor\\" style=\\"display: inline-block;\\" draggable=\\"true\\"><span style=\\"cursor: move; z-index: 1; border-radius: 10px; color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity: 1;\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill: #aee239;\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill: #8fbe00;\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>rur</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>rur</text></svg><span style=\\"padding-left: 45px;\\">litecoin</span></span></div></div></div></div></div><div class=\\"css-kszx3i\\"><label class=\\"css-gudk1o\\"><input placeholder=\\"Full name\\" pattern=\\"[a-zа-яё]{2,}\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input mask=\\"tether label\\" placeholder=\\"Your tether wallet\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input mask=\\"litecoin label\\" placeholder=\\"Your wallet for litecoin\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><div class=\\"css-75t9v1\\"><label class=\\"\\" for=\\"agree\\"><input id=\\"agree\\" type=\\"checkbox\\"><span>I agree to the processing of personal data</span></label></div><div class=\\"css-1xzpagd\\"><button cursor=\\"not-allowed\\" class=\\"css-zfl023\\"><div class=\\"css-1v1d0dg\\">Create</div></button></div></div></div>"
`;
